packages/components/src/components/Switch/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import styled from 'styled-components';
+import css from '@styled-system/css';
+import { Element } from '../Element';
+
+const SwitchBackground = styled.div(
+  css({
+    width: 7,
+    height: 4,
+    backgroundColor: 'switch.background',
+    borderRadius: 'large',
+    position: 'relative',
+  })
+);
+
+const SwitchToggle = styled.span(
+  css({
+    width: 4,
+    height: 4,
+    backgroundColor: 'switch.foregroundOff',
+    borderRadius: '50%',
+    position: 'absolute',
+    left: 0,
+    transition: 'left ease',
+    transitionDuration: theme => theme.speeds[3],
+  })
+);
+
+const SwitchInput = styled.input(
+  css({
+    width: 0,
+    opacity: 0,
+    position: 'absolute',
+  })
+);
+
+const SwitchContainer = styled(Element)(
+  css({
+    'input:checked + [data-component=SwitchBackground]': {
+      backgroundColor: 'switch.foregroundOn',
+    },
+    'input:checked + [data-component=SwitchBackground] [data-component=SwitchToggle]': {
+      left: theme => theme.space[4] - 3,
+    },
+  })
+);
+
+interface ISwitchProps {
+  on?: boolean;
+  defaultOn?: boolean;
+  onChange?: (event: any) => void;
+}
+
+export const Switch: React.FC<ISwitchProps> = ({ on, defaultOn, ...props }) => (
+  <SwitchContainer as="label">
+    <SwitchInput
```

Comment:
This input doesn't have a label maybe we need something here?

Reply:
The label is given by the label on the left

![Screenshot 2020-01-09 at 8 38 19 AM](https://user-images.githubusercontent.com/1863771/72047758-8af7f300-32bb-11ea-9351-0c6430c157e8.png)

How do we suggest folks to attach them together with for and id, one way could be to make `id` compulsory on the switch and throw a warning if it's not given?

Reply:
Sounds good!